client/consensus/aura/src/import_queue.rs

```diff
@@ -186,6 +186,25 @@ where
 		&mut self,
 		mut block: BlockImportParams<B, ()>,
 	) -> Result<(BlockImportParams<B, ()>, Option<Vec<(CacheKeyId, Vec<u8>)>>), String> {
+		// When importing whole state we don't verify the seal as the state is not available.
+		if block.with_state() {
+			return Ok((block, Default::default()))
+		}
+
+		let info = self.client.info();
+
+		// If we are importing a block in the gap, we skip all checks.
+		//
+		// It is expected that the block after the gap was checked/chosen properly, e.g. by warp
+		// syncing to this block.
+		if info
```

Comment:
Instead of checking for gaps specifically, just check for `block.state_action.skip_execution_checks()`
It returns `true` when all checks that involve calling into parent runtime should be skipped.

Reply:
Might as well remove the same check on line 249 below.